test/AftermarketDevice.test.ts

```diff
@@ -1050,7 +1050,361 @@ describe('AftermarketDevice', function () {
     });
   });
 
-  describe('pairAftermarketDeviceSign', () => {
+  describe('pairAftermarketDeviceSign(uint256,uint256,bytes,bytes)', () => {
+    let claimOwnerSig1: string;
+    let claimAdSig1: string;
+    let pairVehicleSig1: string;
+    let pairAdSig1: string;
+    let pairAdSig2: string;
+    before(async () => {
+      claimOwnerSig1 = await signMessage({
+        _signer: user1,
+        _primaryType: 'ClaimAftermarketDeviceSign',
+        _verifyingContract: aftermarketDeviceInstance.address,
+        message: {
+          aftermarketDeviceNode: '1',
+          owner: user1.address
+        }
+      });
+      claimAdSig1 = await signMessage({
+        _signer: adAddress1,
+        _primaryType: 'ClaimAftermarketDeviceSign',
+        _verifyingContract: aftermarketDeviceInstance.address,
+        message: {
+          aftermarketDeviceNode: '1',
+          owner: user1.address
+        }
+      });
+      pairVehicleSig1 = await signMessage({
+        _signer: user1,
+        _primaryType: 'PairAftermarketDeviceSign',
+        _verifyingContract: aftermarketDeviceInstance.address,
+        message: {
+          aftermarketDeviceNode: '1',
+          vehicleNode: '1'
+        }
+      });
+      pairAdSig1 = await signMessage({
+        _signer: adAddress1,
+        _primaryType: 'PairAftermarketDeviceSign',
+        _verifyingContract: aftermarketDeviceInstance.address,
+        message: {
+          aftermarketDeviceNode: '1',
+          vehicleNode: '1'
+        }
+      });
+      pairAdSig2 = await signMessage({
+        _signer: adAddress2,
+        _primaryType: 'PairAftermarketDeviceSign',
+        _verifyingContract: aftermarketDeviceInstance.address,
+        message: {
+          aftermarketDeviceNode: '2',
+          vehicleNode: '1'
+        }
+      });
+    });
+
+    beforeEach(async () => {
+      await adIdInstance
+        .connect(manufacturer1)
+        .setApprovalForAll(aftermarketDeviceInstance.address, true);
+      await aftermarketDeviceInstance
+        .connect(manufacturer1)
+        .mintAftermarketDeviceByManufacturerBatch(
+          1,
+          mockAftermarketDeviceInfosList
+        );
+      await vehicleInstance
+        .connect(admin)
+        .mintVehicle(1, user1.address, C.mockVehicleAttributeInfoPairs);
+      await aftermarketDeviceInstance
+        .connect(admin)
+        .claimAftermarketDeviceSign(
+          1,
+          user1.address,
```

Comment:
Does the same user own the vehicle id and aftermarket device id? Perhaps we should check the case where the owners are different, since that's the main thrust here.

Reply:
I think I got lost in my own notation. Now seems correct.

Reply:
Always happens with complicated setup.